### ISSUE TITLE
Inheritdoc on constructor raises MA0196

### ISSUE THREAD

**Issue by cbersch** [meziantou/Meziantou.Analyzer#1145]
### Version of the Meziantou.Analyzer NuGet package

3.0.77

### Rule Identifier

MA0196

### Target Framework

net10.0

### C# Language version

C# 14

### Description

MA0196 is about overrides and interface implementations. What about constructors? Yes, these aren't overrides, but a plain `<inheritdoc />` seems to work for them if there is a parent constructor with the same arguments.

### Reproduction Steps

Minimal code:

```c#
public class Child : Parent
{
    /// <inheritdoc />
    public Child(int[] values) : base(values)
    {
    }
}

public class Parent
{
    public int[] Values { get; }

    /// <summary>
    /// This is a summary
    /// </summary>
    /// <param name="values"></param>
    protected Parent(int[] values)
    {
        Values = values;
    }
}
```
This raises a MA0196 in line 3. However, at least Visual Studio seems to interpret this `inheritdoc` as expected:

<img width="378" height="412" alt="Image" src="https://github.com/user-attachments/assets/5f556f8f-5586-40b7-85b2-9b2e0d329aa7" />

When I remove the `inheritdoc`, no summary is shown:

<img width="374" height="345" alt="Image" src="https://github.com/user-attachments/assets/b590e4ba-488f-4685-8c14-369e26cdfa7d" />

**Comment by meziantou** [comment]
I think this one is tricky. The IDE can see the `base` call. However, if you consume the library from a NuGet package, I'm not sure the inheritdocs would work. Indeed, the base ctor is not part of the method signature, but implementation. This means decompiling the code to see which ctor it calls to find the good documentation.
Maybe it would require a flag for IDE vs Documentation generation?

**Comment by meziantou** [comment]
I tried in VS and the logic is a bit different from what I thought. In fact, it matches the parent ctor using the parameter types, not the `base` call. So, it may be ok to use this logic.

**Comment by cbersch** [comment]
That looks good, just check it. Thanks!
